src/mbgl/map/transform_state.cpp

```diff
@@ -161,7 +161,14 @@ double TransformState::getScale() const {
 }
 
 void TransformState::setMinZoom(const double minZoom) {
-    min_scale = zoomScale(minZoom);
+    double zoom = minZoom;
+    if (minZoom > 25.5) {
+        zoom = 25.5;
+    }
+    if (minZoom < 0) {
+        zoom = 0.0;
```

Comment:
This can be succinctly expressed as `util::clamp(minZoom, 0.0, 25.5)`. But you should also ensure that the minimum zoom level is less than the maximum.